projects/fsastorefrontlib/src/cms-components/my-account/policy/policies-chart/policies-chart.component.ts

```diff
@@ -0,0 +1,171 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { EChartsOption } from 'echarts';
+import { PolicyService } from '../../../../core/my-account/facade/policy.service';
+import { Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { PolicyChartDataService } from '../../../../core/my-account/services/policy-chart-data.service';
+
+@Component({
+  selector: 'cx-fs-policies-chart',
+  templateUrl: './policies-chart.component.html',
+})
+export class PoliciesChartComponent implements OnInit, OnDestroy {
+  chartOption: EChartsOption;
+  options: any[];
+  isChartVisible: boolean;
+  selectedFrequency: string;
+  policiesByPaymentFrequency;
+  private subscription = new Subscription();
+
+  constructor(
+    protected policyService: PolicyService,
+    protected policyChartDataService: PolicyChartDataService
+  ) {}
+
+  ngOnInit(): void {
+    this.initChart();
+    this.subscription.add(
+      this.policyService.policies$
+        .pipe(
+          tap(policies => {
+            this.policiesByPaymentFrequency = this.policyChartDataService.groupPoliciesByAttribute(
+              policies,
+              'paymentFrequency'
+            );
+            this.setPaymentFrequencyDropdown();
+            this.setChartSeriesData();
+          })
+        )
+        .subscribe()
+    );
+  }
+
+  initChart() {
+    this.chartOption = {
+      title: {
```

Comment:
It seems for me that with this approach it will be hard for end customers to override stuff.. If someone wanted to change a title.text or fontSize for example he would have to extend component and override whole initChart method with duplicating configuration. Instead of that we could maybe extract configuration into separate file and provide it to config module.. then customers could only override configuration or even part of it, without introducing new component..  also some mapping between component (PoliciesChartComponent) and configuration should exist, so that we can provide different configuration for different components.

Reply:
You mean to create config file which contains const variable like this:
`export const optionsConfig = {
    title: {
        text: 'Purchase Order',
        color: '#000033',
        fontSize: 16
    }
}`
and use it in the component?